### PR TITLE
DOCSP-34822: commitTxn is idempotent

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-34822>
[Staging](https://preview-mongodbrustagir.gatsbyjs.io/golang/DOCSP-34822-commitTXN-idempotency/fundamentals/transactions/#methods:~:text=Retrying%20a%20Transaction)

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/fundamentals/transactions.txt [hunk 1]
```diff
@@ -71,25 +72,42 @@ following table describes these methods:
      - | Starts a new transaction, configured with the given options, on
          this session. Returns an error if there is already
          a transaction in progress for the session. For more
-         information, see the :manual:`manual entry </reference/method/Session.startTransaction/>`.
+         information, see :manual:`startTransaction()
+         </reference/method/Session.startTransaction/>`  in the Server manual.
```
Comment (jordan-smith721) [inline source/fundamentals/transactions.txt:76]: S: I think these sentences would flow a little better when we say "see the <link> guide (or page) in the..." 
Applies to all "For more information..." instances

```suggestion
         information, see the :manual:`startTransaction()
         </reference/method/Session.startTransaction/>` guide in the Server manual.
```

**Review by jordan-smith721 — APPROVED** [review]
LGTM w/ a small suggestion

File: source/fundamentals/transactions.txt [hunk 2]
```diff
@@ -71,25 +72,42 @@ following table describes these methods:
...
-         information, see the :manual:`manual entry </reference/method/Session.commitTransaction/>`.
+         error if there is no active transaction for the session or if the
+         transaction was ended. For more
+         information, see :manual:`commitTransaction()
+         </reference/method/Session.commitTransaction/>` in the Server
+         manual.
+         
+         .. note:: Retrying a Transaction
+            
+            The ``CommitTransaction()`` method is an idempotent function, which
```
Comment (prestonvasquez) [inline source/fundamentals/transactions.txt:101]: CommitTransaction appears idempotent in experimentation, but I can't find server documentation supporting this. Can we assign someone on the server team to review this PR to ensure this is a true statement?

Comment (rustagir) [reply source/fundamentals/transactions.txt:101]: Sure - I misunderstood the ticket. I can ask a server engineer to review and confirm.

**Review by prestonvasquez — APPROVED** [review]

**Pull request merged**